File: Neetcode/Two_Pointers/11_Container_Most.py
height = [1,8,6,2,5,4,8,3,7]

def maxArea(height):
    size = len(height)
    left, right = 0, size - 1
    maxWater = -1

    while (left < right):
        maxWater = max(maxWater, (min(height[left], height[right]) * (right - left) ) )
        if (height[left] <= height[right]):
            left += 1
    
        else:
            right -= 1
        
    return maxWater

print(maxArea(height))



# brute force O(n^2)
def maxArea2(height):
    size = len(height)
    maxWater = -1
    for i in range(size):
        left, right = i, size - 1
        while (left < right):
          water = (min(height[left], height[right])) * (right - left)
          maxWater = max(maxWater, water)
          right -= 1
    
    return maxWater

File: Neetcode/Two_Pointers/11_Container_Most.cpp
#include <bits/stdc++.h>

using namespace std;

int maxArea(vector<int>& height) 
{
  int size = height.size();
  int left = 0, right = size - 1;
  int max_water = -1;
  while (left < right)
  {
    max_water = max(max_water, (min(height[left], height[right]) * (right - left) ) );
    if (height[left] <= height[right]) left++;
    else right--;
  }

  return max_water;
}

int main()
{
  vector<int> height = {1,8,6,2,5,4,8,3,7};
  cout << maxArea(height);
}